libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -1321,14 +1323,22 @@ void ContractCompiler::appendModifierOrFunctionCode()
 
 			stackSurplus = CompilerUtils::sizeOnStack(modifier.parameters());
 			codeBlock = &modifier.body();
+			experimentalFeaturesInside = modifier.sourceUnit().annotation().experimentalFeatures;
 		}
 	}
 
 	if (codeBlock)
 	{
+		std::set<ExperimentalFeature> experimentalFeaturesOutside = m_context.experimentalFeaturesActive();
+		solAssert(experimentalFeaturesInside.has_value(), "experimentalFeaturesInside not initialized");
+		m_context.setExperimentalFeatures(experimentalFeaturesInside.value());
```

Comment:
Wouldn't this also work?
```suggestion
		m_context.setExperimentalFeatures(codeBlock->sourceUnit().annotation().experimentalFeatures);
```

Reply:
Probably :) I didn't realize I could just check the source unit of a block :)

Reply:
Done.

I also removed the assert that checks that the experimental feature status does not change inside because I noticed that `accept()` on an AST node cannot really modify the context.